src/sdk/pynni/nni/compression/torch/speedup/infer_shape.py

```diff
@@ -601,6 +792,35 @@ def batchnorm2d_mask(module_masks, mask):
     module_masks.set_output_mask(output_cmask)
     return input_cmask, output_cmask
 
+def linear_mask(module_masks, mask):
+    """
+    Infer input and output shape from weight mask
+
+    Parameters
+    ----------
+    module_masks : ModuleMasks
+        The ModuleMasks instance of the batchnorm2d
+    mask : dict
+        The mask of its weights, from the user provided mask file
+
+    Returns
+    -------
+    CoarseMask, CoarseMask
+        The mask of its input tensor, the mask of its output tensor
+    """
+    assert 'weight' in mask
+
+    nonzero_index = torch.nonzero(mask['weight'].sum(0), as_tuple=True)[0]
```

Comment:
better to make it clear that this function only deals with 0th dimension, using for example asserts